l10n_it_codici_carica/models/codice_carica.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+
+class CodiceCarica(models.Model):
+    _name = 'codice.carica'
+    _description = 'Codice Carica'
+
+    @api.constrains('code')
+    def _check_code(self):
+        domain = [('code', '=', self.code)]
+        elements = self.search(domain)
+        if len(elements) > 1:
+            raise ValidationError(
+                _("The element with this code already exists"))
+
+    code = fields.Char(string='Code', size=2)
+    name = fields.Char('Name')
```

Comment:
label here and above are redundant, you could consider to avoid them.

As wrritten at https://www.odoo.com/documentation/10.0/reference/orm.html#basic-fields:

> if not set, the ORM takes the field name in the class (capitalized).